samples/lo-device/lo-device.js

```diff
@@ -12,371 +12,479 @@
-    client.publish(topicResource,msgResourceStr);
+  var msgResourceStr = JSON.stringify(deviceResources);
+  logger.info('[' + topicResource + ']> ' + msgResourceStr);
+  client.publish(topicResource, msgResourceStr);
 }
 
 function publishDeviceConfig() {
-    var msgConfigStr = JSON.stringify(deviceConfig);
-    logger.info("["+topicConfig+"]> " + msgConfigStr);
-    client.publish(topicConfig,msgConfigStr);
+  var msgConfigStr = JSON.stringify(deviceConfig);
+  logger.info('[' + topicConfig + ']> ' + msgConfigStr);
+  client.publish(topicConfig, msgConfigStr);
 }
 
 function publishDeviceData() {
-    var msgDataStr = JSON.stringify(getDeviceDataMessage());
-    logger.info("["+topicData+"]> " + msgDataStr);
-    client.publish(topicData,msgDataStr);
+  var msgDataStr = JSON.stringify(getDeviceDataMessage());
+  logger.info('[' + topicData + ']> ' + msgDataStr);
+  client.publish(topicData, msgDataStr);
 }
 
-function downloadFile(resourceId, resourceNewVersion, resourceUrl, resourceSize, resourceMd5, downloadError, endDownloadCb) {
-    logger.debug("download resource "+ resourceId + " version " + resourceNewVersion + " from " + resourceUrl);
-    var file = fs.createWriteStream("lastfirmware.raw");
-    http.get(resourceUrl, function(response) {
-        response.pipe(file);
-        file.on('finish', function() {
-              logger.debug("download done");
-              file.close(endDownloadCb);  // close() is async, call endDownloadCb after close completes.
-        });
-        if (downloadError) {
-          setTimeout(function() {
-             logger.debug("this is an error while downloading firmware: enforce exit()");
-             process.exit();
-          }, 10)
-        }
-    });
+function sleep(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
 }
 
-function handleConfigUpdate(message) {
-
-    var request = JSON.parse(message);
-    var jsonRequest = JSON.stringify(request);
-
-    logger.info("<["+topicConfigUpdate+"] ", jsonRequest);
-    if (configFailure > 0) {
-
-        // hack a wrong value for a requested parameter
-        var msgWrongCfg = request;
-        var firstParam = Object.keys(request.cfg)[0];
-        msgWrongCfg.cfg[firstParam]["v"] = 666;
-
-        var msgWrongCfgStr = JSON.stringify(msgWrongCfg);
-        logger.info("FAILED config ["+topicConfig+"]> " + msgWrongCfgStr);
-        client.publish(topicConfig, msgWrongCfgStr);
-        configFailure--;
-
-    } else { // success
+async function generateGeoloc(geolocs, nb = 1) {
+  for (let index = 0; index < nb; index++) {
+    deviceUrn =
+      'urn:lo:nsid:' +
+      deviceNamespace +
+      ':' +
+      deviceId +
+      '_' +
+      Math.ceil(Math.random() * 100000000000);
+
+    var count = connectionCount;
+    forceReconnect();
+    while (count === connectionCount) {
+      await sleep(500);
+    }
+    await sleep(1000);
+    geoloc = [
+      geolocs[1][0] + (geolocs[0][0] - geolocs[1][0]) * Math.random(),
```

Comment:
suggest: 
- add a brief comment to understand this ? or use intermediate well named function? 
- check requirement on input parameter else throw err ?

Reply:
done

Reply:
🙏